command/agent/keyring_endpoint.go

```diff
@@ -77,6 +78,67 @@ func (s *HTTPServer) JWKSRequest(resp http.ResponseWriter, req *http.Request) (a
 	return out, nil
 }
 
+// OIDCDiscoveryRequest implements the OIDC Discovery protocol for using
+// workload identity JWTs with external services.
+//
+// See https://openid.net/specs/openid-connect-discovery-1_0.html for details.
+func (s *HTTPServer) OIDCDiscoveryRequest(resp http.ResponseWriter, req *http.Request) (any, error) {
+	if req.Method != http.MethodGet {
+		return nil, CodedError(405, ErrInvalidMethod)
+	}
+
+	args := structs.GenericRequest{}
+	if s.parse(resp, req, &args.Region, &args.QueryOptions) {
+		return nil, nil
+	}
+
+	conf := s.agent.GetConfig()
+
+	//FIXME(schmichael) should we bother implementing an RPC just to get region
+	//forwarding? I think *not* since consumers of this endpoint are code that is
+	//intended to be talking to a specific region directly.
+	if args.Region != conf.Region {
+		return nil, CodedError(400, "Region mismatch")
+	}
```

Comment:
This is a HTTP endpoint, so clients expose it too, right? Don't we need RPC forwarding for that to work?

Reply:
No because there's no RPC involved at all: any _agent_ regardless of whether it's client or server can serve this request...

...but for operational simplicity I should probably make it an RPC and server-only. That way you only have to configure the `issuer` on your servers, and we can avoid having a situation where every agent in a cluster must have the exact same config value or risk breaking something.